src/app/services/to-do-list-service/to-do-list.service.ts

```diff
@@ -65,24 +84,103 @@ export class ToDoListService {
         this._editedItemId = null;
     }
 
-    public addItem({ title, description }: Omit<ToDoListItem, 'id'>) {
-        const newItemId = this.itemIds.length > 0 ? Math.max(...this.itemIds) + 1 : 0;
+    public toggleItemStatus(id: string) {
+        const item = this.getItemById(id);
+        if (!item) {
+            return;
+        }
+
+        let newStatus: ToDoListItemStatus;
+        if (item.status === STATUS_OPTIONS.completed) {
+            newStatus = STATUS_OPTIONS.inProgress;
+        } else {
+            newStatus = STATUS_OPTIONS.completed;
+        }
+        this.patchItem({ id, status: newStatus });
+    }
 
-        this._toDoList.push({
-            id: newItemId,
-            title: title.trim(),
-            description: description?.trim(),
+    public getToDoList() {
+        this.setIsLoading(true);
+
+        this._api.getItems().subscribe({
```

Comment:
отписку бы

Reply:
Тут тоже прошу помощи. Вроде как здесь не нужна отписка:

`You don't have to unsubscribe from HttpClient observables because they unsubscribe automatically after the server request responds or times out.`
https://v17.angular.io/guide/http-request-data-from-server